middleman_protocol/tests/tests.py

```diff
@@ -0,0 +1,187 @@
+        raw_message_without_separator = AbstractMiddlemanMessage.remove_frame_separator(raw_message)
+        deserialized_message = AbstractMiddlemanMessage.deserialize(
+            raw_message_without_separator,
+            public_key=CONCENT_PUBLIC_KEY,
+            is_frame_separator_expected=False,
+        )
+
+        assert isinstance(deserialized_message, type(payload))
+        assert deserialized_message == payload
+
+    @pytest.mark.parametrize(('middleman_message_type', 'payload'), [
+        (GolemMessageMiddlemanMessage,            Ping()),
+        (ErrorMiddlemanMessage,                   (111, 'error_message')),
+        (AuthenticationChallengeMiddlemanMessage, b'random_bytes'),
+        (AuthenticationResponseMiddlemanMessage,  b'TODO'),
+    ])  # pylint: disable=no-self-use
+    def test_that_sending_message_over_tcp_socket_should_preserve_original_data(
+        self,
+        middleman_message_type,
+        payload,
+    ):
+        middleman_message = middleman_message_type(payload, self.request_id)
+        raw_message = middleman_message.serialize(private_key=CONCENT_PRIVATE_KEY)
+
+        with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as server_socket:
+            with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as client_socket:
+                server_socket.bind(('127.0.0.1', 8001))  # pylint: disable=no-member
+                server_socket.listen(1)  # pylint: disable=no-member
+
+                client_socket.connect(('127.0.0.1', 8001))  # pylint: disable=no-member
+
+                (connection, _address) = server_socket.accept()  # pylint: disable=no-member
+
+                client_socket.send(raw_message)  # pylint: disable=no-member
+                deserialized_message = next(handle_message_receive(connection, CONCENT_PUBLIC_KEY))
+
+        assert isinstance(deserialized_message, type(payload))
+        assert deserialized_message == payload
+
+    def test_that_serializing_different_golem_message_middleman_messages_should_keep_part_of_header_the_same(self):
+        message_1 = GolemMessageMiddlemanMessage(
+            ServiceRefused(reason=ServiceRefused.REASON.InvalidRequest),
+            self.request_id
+        ).serialize(
+            private_key=CONCENT_PRIVATE_KEY
+        )
+        message_2 = GolemMessageMiddlemanMessage(Ping(), self.request_id).serialize(private_key=CONCENT_PRIVATE_KEY)
+
+        assert message_1[65:70] == message_2[65:70]
+
+    def test_that_serializing_different_messages_with_the_same_payload_length_should_place_payload_length_on_the_same_bytes(self):
+        message_1 = AuthenticationChallengeMiddlemanMessage(b'test', self.request_id).serialize(private_key=CONCENT_PRIVATE_KEY)
+        message_2 = AuthenticationResponseMiddlemanMessage(b'test', self.request_id * 1000).serialize(private_key=CONCENT_PRIVATE_KEY)
+
+        assert message_1[70:74] == message_2[70:74]
```

Comment:
:+1: done

I use `assert` mostly because I found them more readable then `assertpy` statements. It is not possible to use pytest along with `unittest.TestCase` unfortunately. But yes, I agree.